### ISSUE TITLE
Conflit with AuthMe Reloaded

### ISSUE THREAD

**Issue by Blamo27** [elBukkit/MagicPlugin#45]
Bug with Authme and the wand, what the fuck ...
https://streamable.com/twis

The inventory is different in the first world and the second (/warp pvpbox), which allows me to gain the number of SP I have as levels.
https://streamable.com/2hov

**Comment by NathanWolf** [comment]
Set

```
mana_display: none
sp_display: none
```

In config.yml to prevent this.

**Comment by Blamo27** [comment]
How can the players see their mana and SP then ?

**Comment by NathanWolf** [comment]
You could install Featherboard and configure it to show mana and SP. SP is also shown in the spell shop's title.

Otherwise that's it, and no offense to AuthMe but I don't really support cracked servers, sorry. Also don't really need that kind of language in an issue tracker :\

**Comment by Blamo27** [comment]
Sorry about that

**Comment by sgdc3** [comment]
@NathanWolf Authme can be useful also on online mode servers, it can add an additional security layer ;)

**Comment by Blamo27** [comment]
Also, the same issue occurs simply with multiverse and separate inventories, as shown in the second video

**Comment by sgdc3** [comment]
@NathanWolf The authme issue can be fixed very quickly: just decrase your listeners priority.

**Comment by NathanWolf** [comment]
@sgdc3 Good point, fair enough :)

In general Magic has issues with any plugin that directly inspects or modifies a player's XP. 

Maybe an event priority change could help- I thought I had my quit events on highest and join events on lowest though (meaning it tries to restore real XP before anything else on quit, and turns on wands after anything else). 

I assume AuthMe is saving/restoring XP on join/quit?

**Comment by Blamo27** [comment]
@NathanWolf  And then for multiple reconnect that destroy the wand ?
Any fix ? :(

> https://streamable.com/twis

**Comment by NathanWolf** [comment]
Oh- that is going to be a problem. Not sure what plugin's doing that, but it's not something I can fix. That means a plugin is saving a player's inventory without complete data (usually means it's not using the Bukkit serialization API to save items). It is throwing away the NBT data which effectively destroys magic items.

**Comment by NathanWolf** [comment]
I should also mention that this will happen with ANY plugin that saves inventories prior to 1.8.6, unless they hook into vanilla item serialization, since Spigot had no support for saving custom NBT data until then. Again, no fix if that's the case, unfortunately.

**Comment by sgdc3** [comment]
No, authme doesn't touch any player data since version 5.1  ;)

**Comment by sgdc3** [comment]
yeah item serialisation is broken since 1.8.5, we had to use protocollib to hide the inventory with authme :(

**Comment by Blamo27** [comment]
So It's due to ProtocolLib ?

**Comment by Blamo27** [comment]
@NathanWolf Are you sure that your backup_player_inventory is not the problem ?
> https://streamable.com/twis

**Comment by NathanWolf** [comment]
Pretty sure, yes. You can turn that off i you're quetioning it- that only kicks in if the server crashes, and you'd see messages in your server logs about it- something like "Restoring PLAYER's inventory, did the server crash?"

Are you on a spigot version before or after 1.8.6?

**Comment by Blamo27** [comment]
@NathanWolf Yes, so is it caused by ProtocolLib ?

**Comment by NathanWolf** [comment]
I doubt it is caused by protocol lib.

Do you have any plugins that are saving and restoring player inventory items?

**Comment by Blamo27** [comment]
MultiInv, MobArena, AuthMe (ProtocolLib), but the bug is caused while reconnecting without loging in (actually typing your password)

**Comment by NathanWolf** [comment]
I assume it's either MultiInv or AuthMe or some interaction between the two.  If possible, do some testing with just MultiInv- that sounds like the suspicious one, assuming what it does is swap out player inventories per-world.

**Comment by Blamo27** [comment]
I don't think it can be MultiInv, because previously we had Multiverse-Inventories and the bug was there already, the bug still occurs even if we changed to MultiInv.

**Comment by NathanWolf** [comment]
I was pretty certain that MV-Inventories handled serialization ok.

I'm not certain from @sgdc3 's responses whether or not Authme stores/restores player inventories on logic- I would have to think yes, given how it works?

**Comment by sgdc3** [comment]
@NathanWolf No, authme 5.X is designed to don't touch any inventory related stuff.

**Comment by NathanWolf** [comment]
@sgdc3 And you mentioned using protocol lib to fix/improve item serialization? Could you provide any more details there?

What we are looking for is some way that the custom NBT data on an item would get lost. Most usually this is due to a plugin serializing the item with custom code, not using the Bukkit API, and not accounting for unknown NBT tags.

**Comment by sgdc3** [comment]
Authme doesn't use serialization, we use plib to send a fake inventory to the client

**Comment by Blamo27** [comment]
The bug is caused by AuthMe, I just tested in a temporary server :
> 149.202.70.149:41485

The wand is completely destroyed, just like in my main server.

**Comment by sgdc3** [comment]
@Blamo27 Version number? (authme)

**Comment by Blamo27** [comment]
AuthMe 5.0 - Beta 3

**Comment by NathanWolf** [comment]
@Blamo27 are you or your players using creative mode?

**Comment by sgdc3** [comment]
That's the problem xD Beta 3 is very glitched. I wanted to remove the download link from the spigotmc page but i don't have permissions.

**Comment by Blamo27** [comment]
I'm using creative mode, the users aren't.
And they loose their wand.

**Comment by sgdc3** [comment]
https://circleci-tkn.rhcloud.com/api/v1/project/AuthMe-Team/AuthMeReloaded/tree/master/latest/artifacts/AuthMe.jar <-- latest dev build

**Comment by Blamo27** [comment]
But the latests version of AuthMe is buggy.

**Comment by sgdc3** [comment]
It should be stable now.

**Comment by sgdc3** [comment]
Our open critical issues: https://github.com/Xephi/AuthMeReloaded/issues?q=is%3Aissue+is%3Aopen+label%3Acritical (most of them are fixed but we are waiting for an answer from the reporter)

**Comment by Blamo27** [comment]
Thanks @sgdc3 and @NathanWolf
But is this bug fixed ?
> https://streamable.com/2hov

**Comment by sgdc3** [comment]
@Blamo27 If the cause was authme it should be fixed ;)

**Comment by sgdc3** [comment]
just try it on a  local server

**Comment by Blamo27** [comment]
I don't think so, It's a SP glitch of Magic Plugins

**Comment by NathanWolf** [comment]
If you have plugins that interact with player experience, you will still need to turn off th SP and mana displays. This is likely a separate issue, doesn't sound like authme would cause that.

**Comment by Blamo27** [comment]
Yes this is caused by MultiInv, but we cannot display SP. We need to buy Featherboard ... :/

**Comment by NathanWolf** [comment]
I would consider adding direct action bar support, but only for 1.9 and probably not until after a few more urgent things.

Otherwise, is there a way to turn off the per-world XP, assuming that's what it's doing?

Are the issues with authme resolved?

**Comment by Blamo27** [comment]
AuthMe issue is resolved, thanks, but I think that we cannot turn off the per-world XP,

**Comment by sgdc3** [comment]
@NathanWolf You should use protocollib to display a custom xp bar

**Comment by NathanWolf** [comment]
@sgdc3 That is a genius suggestion - perhaps I will, so long as I can make it option and fall back to current behavior if ProtocolLib is not present :)

**Comment by Blamo27** [comment]
@sgdc3 I just updated AuthMe in my server, and I have some tab bugs (list of players). I can't see any players in the tab, not even myself, and in addition everyone is invisible - litteraly, not with the effect, they do not appear clientside.

**Comment by Blamo27** [comment]
@sgdc3 It's not really stable.

**Comment by sgdc3** [comment]
@Blamo27 Wtf? Are you sure you only updated authme?

**Comment by sgdc3** [comment]
Wait, found the cause: go in the config and remove the hideTablistBeforeLogin option

**Comment by Blamo27** [comment]
No, It's because we did /authme reload, and AuthMe lost connection to MySQL ...

**Comment by sgdc3** [comment]
Are you sure?

**Comment by Blamo27** [comment]
Yes, I'm sure.

**Comment by sgdc3** [comment]
the new reload is still a WIP, if you ignore it everything should run fine

**Comment by sgdc3** [comment]
@Blamo27 Can you provide the log of the reload command?

**Comment by Blamo27** [comment]
Okey, I'll stop reload AuthMe, maybe I should try to use PlugMan
 Yes the server is flooding
> [Craft Scheduler Thread - 26/WARN]: [AuthMe] Error during SQL operation: [SQLException]: HikariDataSource HikariDataSource (AuthMeMYSQLPool) has been closed.

**Comment by NathanWolf** [comment]
Magic 5.5.3 and 6.0.2 both now use a "virtual" mana bar via packets, which should eliminate any compatibility issues with XP and other plugins.

**Comment by Blamo27** [comment]
Thanks.
